File: huawei/2_Transcoding_Network Delivery/test/show_map.py
N = map.N

for i in range(N):
    for j in range(N):
        node = map.node_matrix[i][j]
        print(node.distance, end=" ")
    print()
print("==================================")
for i in range(N):
    for j in range(N):
        node = map.node_matrix[i][j]
        if node.best_direction == "UP":
            print("↑", end=" ")
        elif node.best_direction == "RIGHT":
            print("→", end=" ")            
        elif node.best_direction == "DOWN":
            print("↓", end=" ")            
        elif node.best_direction == "LEFT":
            print("←", end=" ")
        else:
            print("U", end=" ")
    print()

File: huawei/2_Transcoding_Network Delivery/test/show_map.cpp
N = node_map.N

for(int i=0; i<node_map.N; i++){
    for(int j=0; j<node_map.N; j++){
        Node& node = node_map.node_matrix[i][j]
        cout << node.distance << " ";
    }
    cout << endl;
}
cout << "==================================" << endl;

for(int i=0; i<N; i++){
    for(int j=0; j<N; j++){
        Node& node = node_map.node_matrix[i][j]
        cout << node.distance << " ";
        if (node.best_direction == "UP"){
            cout << "↑" << " ";
        }elif (node.best_direction == "UP"){
            cout << "→" << " ";
        }elif (node.best_direction == "UP"){
            cout << "↓" << " ";
        }elif (node.best_direction == "UP"){
            cout << "←" << " ";
        }else{
            cout << "U" << " ";
        }
    }
    cout << endl;
}